src/coreclr/jit/importer.cpp

```diff
@@ -7299,15 +7299,17 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 // fall through
 
             COND_JUMP:
-
+            {
                 /* Fold comparison if we can */
 
                 op1 = gtFoldExpr(op1);
 
                 /* Try to fold the really simple cases like 'iconst *, ifne/ifeq'*/
                 /* Don't make any blocks unreachable in import only mode */
 
-                if (op1->gtOper == GT_CNS_INT)
+                GenTree* effectiveOp1 = op1->gtEffectiveVal();
+
+                if (effectiveOp1->OperIs(GT_CNS_INT))
                 {
```

Comment:
Doesn't this drop the side effects silently now?

Reply:
or rather, just wrap the comma in an unused node so we don't need to worry about chains of commas and then append